operator/pkg/name/name.go

```diff
@@ -52,19 +52,13 @@ const (
 	// Addon root component
 	AddonComponentName ComponentName = "AddonComponents"
 
-	// Legacy addon components
-	PrometheusComponentName ComponentName = "Prometheus"
-	KialiComponentName      ComponentName = "Kiali"
-	GrafanaComponentName    ComponentName = "Grafana"
-	TracingComponentName    ComponentName = "Tracing"
-	CoreDNSComponentName    ComponentName = "Istiocoredns"
-
 	// Operator components
 	IstioOperatorComponentName      ComponentName = "IstioOperator"
 	IstioOperatorCustomResourceName ComponentName = "IstioOperatorCustomResource"
 
 	// Component names used in old versions
 	InjectorComponentName       ComponentName = "Injector"
+	CertManagerComponentName    ComponentName = "CertManager"
```

Comment:
i feel like we should try to get version based naming into 1.5, otherwise we're going to have to do these sort of backward compatibility hacks for 1.6, where more components will be legacy.